projects/plugins/jetpack/extensions/blocks/calendly/test/edit.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * @jest-environment jsdom
+ */
+
+/**
+ * External dependencies
+ */
+import '@testing-library/jest-dom/extend-expect';
+import userEvent from '@testing-library/user-event';
+import { render, screen, waitFor } from '@testing-library/react';
+
+// Need to mock InnerBlocks before import the CalendlyEdit component as it
+// requires the Gutenberg store setup to operate.
+jest.mock( '@wordpress/block-editor', () => ( {
+	...jest.requireActual( '@wordpress/block-editor' ),
+	InnerBlocks: () => <button>Mocked button</button>,
+} ) );
+
+/**
+ * Internal dependencies
+ */
+import testEmbedUrl from '../../../shared/test-embed-url';
+import CalendlyEdit from '../edit';
+
+jest.mock(
+	'../../../shared/test-embed-url',
+	() => ( {
+		__esModule: true,
+		default: jest.fn().mockImplementation( ( url ) => {
+			return new Promise( ( resolve, reject ) => {
+				url === 'https://calendly.com/invalid-url' ? reject() : resolve( url );
+			} );
+		} )
+	} )
+);
+describe( 'CalendlyEdit', () => {
+	const defaultAttributes = {
+		backgroundColor: '#ffffff',
+		hideEventTypeDetails: false,
+		primaryColor: '#7d7d7d',
+		textColor: '#000000',
+		style: 'inline',
+		url: 'https://calendly.com/username',
+	};
+
+	const createErrorNotice = jest.fn();
+	const removeAllNotices = jest.fn();
+	const setAttributes = jest.fn();
+
+	const defaultProps = {
+		attributes: defaultAttributes,
+		setAttributes,
+		className: '',
+		clientId: 1,
+		name: 'jetpack/calendly',
+		noticeOperations: {
+			removeAllNotices,
+			createErrorNotice,
+		},
+	};
+
+	const propsWithoutUrl = {
+		...defaultProps,
+		attributes: {
+			...defaultAttributes,
+			url: '',
+		},
+	};
+
+	beforeEach( () => {
+		createErrorNotice.mockClear();
+		removeAllNotices.mockClear();
+		setAttributes.mockClear();
+	} );
+
+	test( 'validates block attributes', () => {
+		const attributes = { ...defaultAttributes, invalid: true };
+
+		render( <CalendlyEdit { ...{ ...defaultProps, attributes } } /> );
+
+		expect( setAttributes ).toHaveBeenCalledWith( defaultAttributes );
+	} );
+
+	test( 'set undefined url and displays error when invalid url supplied', async () => {
+		const attributes = { ...defaultAttributes, url: 'https://calendly.com/invalid-url' };
+		render( <CalendlyEdit { ...{ ...defaultProps, attributes } } /> );
+
+		expect( testEmbedUrl ).toHaveBeenCalledWith( attributes.url, expect.anything() );
+
+		await waitFor( () => expect( setAttributes ).toHaveBeenCalledWith( { url: undefined } ) );
```

Comment:
when expectation is dependant on promise resolution wrapping the expectation in a `waitFor` will normally be enough to ensure that promise is resolved before the expectation runs.

Reply:
Could have sworn I'd tried `waitFor`, hence the leftover `async` on the test function. Thanks again.